src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidator.java

```diff
@@ -21,6 +45,64 @@ private boolean validateName(String entityName, Errors errs, String loggingConte
                 && StringValidators.validateCharacters(entityName, location, errs, loggingContext);
     }
 
+    private boolean validateIncorporationCountry(String country, Errors errs, String loggingContext) {
+        String location = getEntityFieldName(EntityDto.INCORPORATION_COUNTRY_FIELD);
+        return StringValidators.validateStringNotBlank(country, location, errs, loggingContext);
+    }
+
+    private Errors validatePrincipalAddress(AddressDto addressDto, Errors errs, String loggingContext) {
+        String location = getEntityFieldName(EntityDto.PRINCIPAL_ADDRESS_FIELD);
+        addressDtoValidator.validate(location, addressDto, errs, loggingContext);
+        return errs;
+    }
+
+    private boolean validateServiceAddressSameAsPrincipalAddress(Boolean same, Errors errs, String loggingContext) {
+        String location = getEntityFieldName(EntityDto.IS_SERVICE_ADDRESS_SAME_AS_PRINCIPAL_ADDRESS_FIELD);
+        return UtilsValidators.validateNotNull(same, location, errs, loggingContext);
+    }
+
+    private Errors validateServiceAddress(AddressDto addressDto, Errors errs, String loggingContext) {
```

Comment:
Probably we should have one `validateAddress` instead of separating it for `principal` and `service`, what do you think?

Reply:
Just looked at implementation of both methods - they seem to be identical apart from the field name being used, so that could be passed in as an additional parameter and a single method would then suffice, no?

Reply:
Updated this.